File: mouse.js
function mouseClass() {
    var mouse = {};
    //mouse public member variables
    mouse.x = -1;
    mouse.y = -1;
    mouse.indicatorUpDown = -1; //up 1, down 0
    //mouse public member methods
    mouse.setPos = function (t_x, t_y) { mouse.x = t_x; mouse.y = t_y; };
    mouse.setIndicatorUpDown = function (t_updown) { mouse.indicatorUpDown = t_updown; };
    mouse.mouseEvent = function (handler, ev) {
        var t_x, t_y;
        // Get the mouse position relative to the canvas element.
        if (ev.layerX || ev.layerX == 0) { // Firefox
            t_x = ev.layerX;
            t_y = ev.layerY;
        } else if (ev.offsetX || ev.offsetX == 0) { // Opera
            t_x = ev.offsetX;
            t_y = ev.offsetY;
        }
        handler(t_x, t_y);
    };
    mouse.onMouseDown = function (t_x, t_y) {
        mouse.setIndicatorUpDown(0);
        mouse.setPos(t_x, t_y);
    };
    mouse.onMouseUp = function (t_x, t_y) {
        mouse.setIndicatorUpDown(1);
        mouse.setPos(t_x, t_y);
    };
    mouse.onMouseMove = function (t_x, t_y) {
        mouse.setPos(t_x, t_y);
    };
    mouse.reset = function () {
        mouse.indicatorUpDown = -1; //up 1, down 0
    }
    return mouse;
}

File: mouse.cpp
#include "globalSetting.h"

#include "mouse.h"
#include <GL/glfw.h>
#include <stdio.h>

Mouse::Mouse(int& t_screenWidth, int& t_screenHeight) 
{ 
	x = -1; 
	y = -1; 
	indicatorUpDown=-1;

	screenWidth=t_screenWidth;
	screenHeight=t_screenHeight;

	screenX=0.0;
	screenY=0.0;
}
Mouse::~Mouse()
{
}
void Mouse::updateScreenCoordinate(bool pixelCoodinate)
{
	if(pixelCoodinate)
	{
		screenX=x;
	    screenY=WIN_HEIGHT-y;
	}
	else
	{
		screenX=(float)x/((float)screenWidth)*2-1;
		screenY=-(float)y/((float)screenHeight)*2+1;
	}

}
void Mouse::setPos (int t_x, int t_y) 
{ 
	x = t_x; 
	y = t_y; 
	//updateScreenCoordinate();
}
void Mouse::setIndicatorUpDown(int t_updown) 
{ 
	indicatorUpDown = t_updown; 
}
void Mouse::updateMouseStatus() 
{ 
	glfwGetMousePos(&x, &y);
	if(glfwGetMouseButton(GLFW_MOUSE_BUTTON_1)==1)
		onMouseDown(x, y) ;
	else if (glfwGetMouseButton(GLFW_MOUSE_BUTTON_1)==0)
		onMouseUp(x, y) ;

	updateScreenCoordinate(true);
	//printf("xpos %d  ypos %d \n", x, y);
	//printf("xpos %f  ypos %f \n", screenX, screenY);
	//printf("MouseButton %d \n", indicatorUpDown); 
}
void Mouse::onMouseDown(int t_x, int t_y) 
{
	setIndicatorUpDown(0);
	setPos(t_x, t_y);
}
void Mouse::onMouseUp(int t_x, int t_y) 
{
	setIndicatorUpDown(1);
	setPos(t_x, t_y);
}
void Mouse::reset () {
	indicatorUpDown = -1; //up 1, down 0
}